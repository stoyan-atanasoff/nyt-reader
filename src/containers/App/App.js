import React, { Component } from 'react';
import { Switch, Route } from 'react-router-dom';
import NewsDetailsPage from '../NewsDetailsPage';
import NewsListPage from '../NewsListPage';
import HomePage from "../HomePage";
import NotFoundPage from '../../components/NotFound';
import Header from "../../components/Header";
import './App.css';
import {Col, Row} from "antd";

class App extends Component {
  render() {
    return (
      <React.Fragment>
        <Header />
        <Row type="flex" justify="center">
          <Col md={16}>
            <Switch>
              <Route exact path="/" component={HomePage} />
              <Route exact path="/news" component={NewsListPage} />
              <Route exact path="/news/:id" component={NewsDetailsPage} />
              <Route path="" component={NotFoundPage} />
            </Switch>
          </Col>
        </Row>
      </React.Fragment>
    );
  }
}

export default App;
