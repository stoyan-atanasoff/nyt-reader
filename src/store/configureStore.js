import { createBrowserHistory } from 'history';
import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import sagas from '../sagas';
import createReducer from '../reducers';
import {routerMiddleware} from "connected-react-router";

export const history = createBrowserHistory();

export default function (initialState = {}) {
  let composeEnhancers = compose;
  const reduxSagaMonitorOptions = {};

  // if (typeof window === 'object') {
  //   if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
  //     composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({});
  // }
  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);
  const middleWares = [sagaMiddleware, routerMiddleware(history)];
  const enhancers = [applyMiddleware(...middleWares)];
  const store = createStore(createReducer(), initialState, composeEnhancers(...enhancers));
  sagaMiddleware.run(sagas);

  return store;
}
