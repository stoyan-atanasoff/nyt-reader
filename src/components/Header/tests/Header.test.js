import React from 'react';
import {render, fireEvent } from "@testing-library/react";
import {Provider} from "react-redux";
import {history} from "../../../store/configureStore";
import {ConnectedRouter} from "connected-react-router";
import App from "../../../containers/App";

import configureStoreOrig from '../../../store/configureStore';
let store = configureStoreOrig();

describe('Header', () => {

  it('Header should render and match the snapshot', async () => {
    const {
      container
    } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <App />
        </ConnectedRouter>
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should open drawer menu', () => {
    const openDrawerButton = document.body.querySelector('button.bars-menu');
    fireEvent.click(openDrawerButton);
    expect(document.body.innerHTML).toMatchSnapshot();
  });

  it('Should close drawer menu', async () => {
    const closeDrawerButton = document.body.querySelector('button.ant-drawer-close');
    fireEvent.click(closeDrawerButton);
    expect(document.body.innerHTML).toMatchSnapshot();
  })

  it('Should drawer link works', async () => {
    const drawerLink = document.body.querySelector('a.drawer-link ');
    fireEvent.click(drawerLink);
    expect(document.body.innerHTML).toMatchSnapshot();
  })
});
