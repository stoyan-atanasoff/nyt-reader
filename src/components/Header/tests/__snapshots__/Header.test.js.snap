// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header should render and match the snapshot 1`] = `
<div
  class="ant-row"
>
  <div
    class="nav-bar"
  >
    <div
      class="ant-col ant-col-xs-0 ant-col-md-24"
    >
      <a
        class="header-link selected"
        href="/"
      >
        Home
      </a>
      <a
        class="header-link "
        href="/news"
      >
        News
      </a>
    </div>
    <div
      class="ant-col ant-col-xs-24 ant-col-md-0"
    >
      <button
        class="ant-btn bars-menu ant-btn-primary ant-btn-round ant-btn-lg ant-btn-icon-only"
        type="button"
      >
        <i
          aria-label="icon: menu"
          class="anticon anticon-menu"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="menu"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </i>
      </button>
    </div>
  </div>
</div>
`;

exports[`Header Should close drawer menu 1`] = `"<div><div class=\\"ant-row\\"><div class=\\"nav-bar\\"><div class=\\"ant-col ant-col-xs-0 ant-col-md-24\\"><a class=\\"header-link selected\\" href=\\"/\\">Home</a><a class=\\"header-link \\" href=\\"/news\\">News</a></div><div class=\\"ant-col ant-col-xs-24 ant-col-md-0\\"><button type=\\"button\\" class=\\"ant-btn bars-menu ant-btn-primary ant-btn-round ant-btn-lg ant-btn-icon-only\\" ant-click-animating-without-extra-node=\\"false\\"><i aria-label=\\"icon: menu\\" class=\\"anticon anticon-menu\\"><svg viewBox=\\"64 64 896 896\\" class=\\"\\" data-icon=\\"menu\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z\\"></path></svg></i></button></div></div></div><div class=\\"ant-row-flex ant-row-flex-center\\"><div class=\\"ant-col ant-col-md-16\\"><div class=\\"homepage-body\\"><span class=\\"homepage-greeting\\">Hello Everyone!</span><br>This is a sample reader client of New York Times top articles! <br>Don't forget to set your own api key.<br><br>This application s built with:<ul><li>React (16.8.6)</li><li>Redux (4.0.1)</li><li>React-router (5.0.1)</li><li>Saga (1.0.3)</li><li>Antd (3.19.4)</li></ul></div></div></div></div><div><div tabindex=\\"-1\\" class=\\"ant-drawer ant-drawer-right\\"><div class=\\"ant-drawer-mask\\"></div><div class=\\"ant-drawer-content-wrapper\\" style=\\"width: 256px; transform: translateX(100%);\\"><div class=\\"ant-drawer-content\\"><div class=\\"ant-drawer-wrapper-body\\" style=\\"overflow: auto; height: 100%;\\"><div class=\\"ant-drawer-header\\"><div class=\\"ant-drawer-title\\">Navigation</div><button aria-label=\\"Close\\" class=\\"ant-drawer-close\\"><i aria-label=\\"icon: close\\" class=\\"anticon anticon-close\\"><svg viewBox=\\"64 64 896 896\\" class=\\"\\" data-icon=\\"close\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z\\"></path></svg></i></button></div><div class=\\"ant-drawer-body\\"><a class=\\"drawer-link selected\\" href=\\"/\\">Home</a><a class=\\"drawer-link \\" href=\\"/news\\">News</a></div></div></div></div></div></div>"`;

exports[`Header Should drawer link works 1`] = `"<div><div class=\\"ant-row\\"><div class=\\"nav-bar\\"><div class=\\"ant-col ant-col-xs-0 ant-col-md-24\\"><a class=\\"header-link selected\\" href=\\"/\\">Home</a><a class=\\"header-link \\" href=\\"/news\\">News</a></div><div class=\\"ant-col ant-col-xs-24 ant-col-md-0\\"><button type=\\"button\\" class=\\"ant-btn bars-menu ant-btn-primary ant-btn-round ant-btn-lg ant-btn-icon-only\\" ant-click-animating-without-extra-node=\\"false\\"><i aria-label=\\"icon: menu\\" class=\\"anticon anticon-menu\\"><svg viewBox=\\"64 64 896 896\\" class=\\"\\" data-icon=\\"menu\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z\\"></path></svg></i></button></div></div></div><div class=\\"ant-row-flex ant-row-flex-center\\"><div class=\\"ant-col ant-col-md-16\\"><div class=\\"homepage-body\\"><span class=\\"homepage-greeting\\">Hello Everyone!</span><br>This is a sample reader client of New York Times top articles! <br>Don't forget to set your own api key.<br><br>This application s built with:<ul><li>React (16.8.6)</li><li>Redux (4.0.1)</li><li>React-router (5.0.1)</li><li>Saga (1.0.3)</li><li>Antd (3.19.4)</li></ul></div></div></div></div><div><div tabindex=\\"-1\\" class=\\"ant-drawer ant-drawer-right\\"><div class=\\"ant-drawer-mask\\"></div><div class=\\"ant-drawer-content-wrapper\\" style=\\"width: 256px; transform: translateX(100%);\\"><div class=\\"ant-drawer-content\\"><div class=\\"ant-drawer-wrapper-body\\" style=\\"overflow: auto; height: 100%;\\"><div class=\\"ant-drawer-header\\"><div class=\\"ant-drawer-title\\">Navigation</div><button aria-label=\\"Close\\" class=\\"ant-drawer-close\\"><i aria-label=\\"icon: close\\" class=\\"anticon anticon-close\\"><svg viewBox=\\"64 64 896 896\\" class=\\"\\" data-icon=\\"close\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z\\"></path></svg></i></button></div><div class=\\"ant-drawer-body\\"><a class=\\"drawer-link selected\\" href=\\"/\\">Home</a><a class=\\"drawer-link \\" href=\\"/news\\">News</a></div></div></div></div></div></div>"`;

exports[`Header Should open drawer menu 1`] = `"<div><div class=\\"ant-row\\"><div class=\\"nav-bar\\"><div class=\\"ant-col ant-col-xs-0 ant-col-md-24\\"><a class=\\"header-link selected\\" href=\\"/\\">Home</a><a class=\\"header-link \\" href=\\"/news\\">News</a></div><div class=\\"ant-col ant-col-xs-24 ant-col-md-0\\"><button type=\\"button\\" class=\\"ant-btn bars-menu ant-btn-primary ant-btn-round ant-btn-lg ant-btn-icon-only\\" ant-click-animating-without-extra-node=\\"false\\"><i aria-label=\\"icon: menu\\" class=\\"anticon anticon-menu\\"><svg viewBox=\\"64 64 896 896\\" class=\\"\\" data-icon=\\"menu\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z\\"></path></svg></i></button></div></div></div><div class=\\"ant-row-flex ant-row-flex-center\\"><div class=\\"ant-col ant-col-md-16\\"><div class=\\"homepage-body\\"><span class=\\"homepage-greeting\\">Hello Everyone!</span><br>This is a sample reader client of New York Times top articles! <br>Don't forget to set your own api key.<br><br>This application s built with:<ul><li>React (16.8.6)</li><li>Redux (4.0.1)</li><li>React-router (5.0.1)</li><li>Saga (1.0.3)</li><li>Antd (3.19.4)</li></ul></div></div></div></div><div><div tabindex=\\"-1\\" class=\\"ant-drawer ant-drawer-right ant-drawer-open\\"><div class=\\"ant-drawer-mask\\"></div><div class=\\"ant-drawer-content-wrapper\\" style=\\"width: 256px;\\"><div class=\\"ant-drawer-content\\"><div class=\\"ant-drawer-wrapper-body\\" style=\\"overflow: auto; height: 100%;\\"><div class=\\"ant-drawer-header\\"><div class=\\"ant-drawer-title\\">Navigation</div><button aria-label=\\"Close\\" class=\\"ant-drawer-close\\"><i aria-label=\\"icon: close\\" class=\\"anticon anticon-close\\"><svg viewBox=\\"64 64 896 896\\" class=\\"\\" data-icon=\\"close\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z\\"></path></svg></i></button></div><div class=\\"ant-drawer-body\\"><a class=\\"drawer-link selected\\" href=\\"/\\">Home</a><a class=\\"drawer-link \\" href=\\"/news\\">News</a></div></div></div></div></div></div>"`;
